packages/main/src/files/file-manager.ts

```diff
@@ -42,23 +42,36 @@ export interface UploadMetadata {
  * @public
  */
 export class GoogleAIFileManager {
+  private _requestOptions: RequestOptions;
   constructor(
     public apiKey: string,
-    private _requestOptions?: RequestOptions,
-  ) {}
+    requestOptions?: RequestOptions,
+  ) {
+    this._requestOptions = {};
+    if (requestOptions) {
+      Object.assign(this._requestOptions, requestOptions);
+      delete this._requestOptions.abortSignal;
+    }
+  }
 
   /**
    * Upload a file
    */
   async uploadFile(
     filePath: string,
     fileMetadata: FileMetadata,
+    requestOptions?: RequestOptions,
   ): Promise<UploadFileResponse> {
     const file = readFileSync(filePath);
+    const filesRequestOptions = Object.create(this._requestOptions);
```

Comment:
I think you can `const filesRequestOptions = {...this._requestOptions, ...requestOptions}`? I think you can set the requestOptions param to default to `{}` (`requestOptions: RequestOptions = {}`) so the second spread will never fail.

Reply:
Done.